packages/modules/fulfillment/src/utils/events.ts

```diff
@@ -0,0 +1,515 @@
+import {
+  Fulfillment,
+  FulfillmentSet,
+  GeoZone,
+  ServiceZone,
+  ShippingOption,
+  ShippingOptionRule,
+  ShippingOptionType,
+} from "@models"
+import { Context, EventBusTypes } from "@medusajs/types"
+import { CommonEvents, FulfillmentUtils, Modules } from "@medusajs/utils"
+
+export function buildFulfillmentAddressEvents({
```

Comment:
This is a lot of code to build out events, I am fine with it for now, but we'll really need to improve how the events are constructed and sent.

Reply:
yes could you provide an example of improvemens you think about?

Reply:
@adrien2p I am thinking a generic `buildEvents` function, as a lot of the `build*` functions do more or less the same thing, with just different parameters.

Reply:
all the properties I am providing here are specific to eac action, I can't abstract them away